[watchOS 9] Support building WebKit

emw-apple opened the pull request:
#### 88f9f7427b6187ddb508a0f2cebfedd44845ebab
<pre>
[watchOS 9] Support building WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=245094">https://bugs.webkit.org/show_bug.cgi?id=245094</a>

Reviewed by Jonathan Bedard.

* Source/WTF/wtf/PlatformHave.h: Set HAVE_UIKIT_BAR_BUTTON_LAYOUT_CUSTOMIZATION for watchOS and tvOS.
* Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
(-[WebAVPlayerViewController tryToStartPictureInPicture]): This method
  is annotated as noreturn for watchOS, but didn&apos;t call
  ASSERT_NOT_REACHED like many of its sibling methods do. Add the
  assertion, as it truly does not appear to be reachable.

Clean up framework stubs for watchOS versions we no longer build for.
Copy over needed stubs for watchOS 9. The only new framework we stub out
is PepperUICore.framework.

* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/AssertionServices.framework/AssertionServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/AuthKit.framework/AuthKit.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/CorePDF.framework/CorePDF.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/DeviceIdentity.framework/DeviceIdentity.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/FileProvider.framework/FileProvider.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/IOKit.framework/IOKit.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/LocalAuthentication.framework/LocalAuthentication.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/PepperUICore.framework/PepperUICore.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/AppSupport.framework/AppSupport.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/AssertionServices.framework/AssertionServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/AuthKit.framework/AuthKit.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/CorePrediction.framework/CorePrediction.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/DeviceIdentity.framework/DeviceIdentity.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/FileProvider.framework/FileProvider.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/GraphicsServices.framework/GraphicsServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/IOKit.framework/IOKit.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/LocalAuthentication.framework/LocalAuthentication.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/PepperUICore.framework/PepperUICore.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/RunningBoardServices.framework/RunningBoardServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/URLFormatting.framework/URLFormatting.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/AppSupport.framework/AppSupport.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/AppSupport.framework/AppSupport.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/CorePrediction.framework/CorePrediction.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/CorePrediction.framework/CorePrediction.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/GraphicsServices.framework/GraphicsServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/GraphicsServices.framework/GraphicsServices.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/PepperUICore.framework/PepperUICore.tbd: Added.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/PrototypeTools.framework/PrototypeTools.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/PrototypeTools.framework/PrototypeTools.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/RunningBoardServices.framework/RunningBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/RunningBoardServices.framework/RunningBoardServices.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/URLFormatting.framework/URLFormatting.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/URLFormatting.framework/URLFormatting.tbd.

Canonical link: <a href="https://commits.webkit.org/254449@main">https://commits.webkit.org/254449@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ed699bb899ac95a3977a87c78e66cc893ea29c26

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88972 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33536 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19845 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98255 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/154634 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/92969 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32041 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/27634 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81337 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92787 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94603 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25448 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75945 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25380 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80315 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80330 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68347 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/80686 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/29832 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/14376 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/74474 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29563 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15356 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26300 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3118 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33002 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38298 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/77335 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31692 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34454 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17133 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
```diff
@@ -483,6 +483,10 @@ - (void)removeObserver
 
 - (void)tryToStartPictureInPicture MY_NO_RETURN
 {
+#if PLATFORM(WATCHOS)
```
Nit: Worth checking with the media folks if this should actually be some "does have Picture in Picture", which we tend to prefer over platform checks.

emw-apple replied:
This is how the check is done in other parts of this file (for example in `-startPictureInPicture`, just below).

I don't really understand why clang started catching this noreturn violation now, as it seems like it's always been the case.

emw-apple committed:
[watchOS 9] Support building WebKit
https://bugs.webkit.org/show_bug.cgi?id=245094

Reviewed by Jonathan Bedard.

* Source/WTF/wtf/PlatformHave.h: Set HAVE_UIKIT_BAR_BUTTON_LAYOUT_CUSTOMIZATION for watchOS and tvOS.
* Source/WebCore/platform/ios/VideoFullscreenInterfaceAVKit.mm:
(-[WebAVPlayerViewController tryToStartPictureInPicture]): This method
  is annotated as noreturn for watchOS, but didn't call
  ASSERT_NOT_REACHED like many of its sibling methods do. Add the
  assertion, as it truly does not appear to be reachable.

Clean up framework stubs for watchOS versions we no longer build for.
Copy over needed stubs for watchOS 9. The only new framework we stub out
is PepperUICore.framework.

* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/AssertionServices.framework/AssertionServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/AuthKit.framework/AuthKit.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/CorePDF.framework/CorePDF.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/DeviceIdentity.framework/DeviceIdentity.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/FileProvider.framework/FileProvider.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/IOKit.framework/IOKit.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/LocalAuthentication.framework/LocalAuthentication.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/PepperUICore.framework/PepperUICore.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/AppSupport.framework/AppSupport.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/AssertionServices.framework/AssertionServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/AuthKit.framework/AuthKit.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/CorePrediction.framework/CorePrediction.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/DeviceIdentity.framework/DeviceIdentity.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/FileProvider.framework/FileProvider.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/GraphicsServices.framework/GraphicsServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/IOKit.framework/IOKit.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/LocalAuthentication.framework/LocalAuthentication.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/PepperUICore.framework/PepperUICore.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/RunningBoardServices.framework/RunningBoardServices.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/URLFormatting.framework/URLFormatting.tbd: Removed.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/AppSupport.framework/AppSupport.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/AppSupport.framework/AppSupport.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/CorePrediction.framework/CorePrediction.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/CorePrediction.framework/CorePrediction.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/GraphicsServices.framework/GraphicsServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/GraphicsServices.framework/GraphicsServices.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/PepperUICore.framework/PepperUICore.tbd: Added.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/PrototypeTools.framework/PrototypeTools.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/7/PrototypeTools.framework/PrototypeTools.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/RunningBoardServices.framework/RunningBoardServices.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/RunningBoardServices.framework/RunningBoardServices.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd.
* WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/9/URLFormatting.framework/URLFormatting.tbd: Renamed from WebKitLibraries/WebKitPrivateFrameworkStubs/watchos/6/URLFormatting.framework/URLFormatting.tbd.

Canonical link: https://commits.webkit.org/254449@main
